code/_core/obj/item/weapon/_weapon.dm

```diff
@@ -107,6 +109,24 @@
 	update_sprite()
 	. = ..()
 
+/obj/item/weapon/clicked_on_by_object(mob/caller,atom/object,location,control,params)
+	if(istype(object,/obj/item/soulgem) && enchantment)
+		var/obj/item/soulgem/G = object
+		if(G.total_charge > 0 && enchantment.charge < enchantment.max_charge)
+			var/chargediff = min(G.total_charge,(enchantment.max_charge - enchantment.charge)) 
+			enchantment.charge += chargediff
+			G.total_charge -= chargediff
+			if(!caller.is_player_controlled())
+				CRASH("How the flying fuck is a mob smart enough to recharge their enchanted weapon![caller.get_debug_name()] did it!")
+			var/mob/living/pcaller = caller //This is stupid and bad.
```

Comment:
In stead of using is_player_controlled, just use is_living()

Reply:
Now uses !is_living(caller)